backend/regalloc/regalloc_split_state.ml

```diff
@@ -631,6 +621,19 @@ let make cfg_with_infos ~next_instruction_id =
     RemoveDominatedSpillsForConstants.optimize cfg_with_infos
       ~destructions_at_end
   in
+  let definitions_at_beginning =
+    Label.Map.filter
+      (fun _ set -> not (Reg.Set.is_empty set))
+      definitions_at_beginning
+  in
+  let destructions_at_end =
+    Label.Map.filter
+      (fun _ (_, set) -> not (Reg.Set.is_empty set))
+      destructions_at_end
+  in
+  let phi_at_beginning =
+    compute_phis cfg_with_infos ~destructions_at_end ~definitions_at_beginning
```

Comment:
The previous version was safe but suboptimal
(generating too many phi, caused by definitions
and destructions that would later be deleted).
Since we had to reason again about names for
this PR, it felt reasonable to treat that minor issue
as well.